django/db/models/query.py

```diff
@@ -145,8 +146,7 @@ class FlatValuesListIterable(BaseIterable):
     def __iter__(self):
         queryset = self.queryset
         compiler = queryset.query.get_compiler(queryset.db)
-        for row in compiler.results_iter():
-            yield row[0]
+        return chain.from_iterable(compiler.results_iter())
```

Comment:
I'd be interested in knowing the reason that makes this faster if you can share @sir-sigurd ?